traveler-swift-core/TravelerKit/Traveler.swift

```diff
@@ -506,6 +505,49 @@ public class Traveler {
         queue.addOperation(fetchOperation)
         OperationQueue.main.addOperation(blockOperation)
     }
+    
+    func fetchItinerary(_ query: ItineraryQuery, completion: @escaping (ItineraryResult?, Error?) -> Void) {
+        guard let travlerId = session.identity else {
+            completion(nil, ItineraryQueryError.unidentifiedTraveler)
+            return
+        }
+        
+        let fetchOperation = AuthenticatedRemoteFetchOperation<ItineraryResult>(path: .itinerary(query, travelerId: travlerId), session: session)
+        let blockOperation = BlockOperation { [unowned fetchOperation] in
+            completion(fetchOperation.resource, fetchOperation.error)
+        }
+
+        blockOperation.addDependency(fetchOperation)
+
+        queue.addOperation(fetchOperation)
+        OperationQueue.main.addOperation(blockOperation)
+    }
+
+    func fetchPurchasedProductDetails(_ query: PurchasedProductDetailsQuery, completion: @escaping (AnyPurchasedItemDetails?, Error?) -> Void) {
```

Comment:
So, I think that we shouldn't be adding a query here. I would be nice to have consistency across our methods, if you're fetching details for an item by sending the item itself (see fetchProductDetails), then it would make sense to try to do the same here. Now, I understand the caveat that, given the way the endpoint is created we need to be aware of an order id. But what I would do then, is to make the call expect both an PurchasedProduct (or item) and an Order. The authpath case would expect both models, (thus solving the routing issue) and it would also force the developer to know that they need an order to get the details of the product. You could even do a check in Core to make sure that the id in the product they're sending exists in one of the products in the order. If it doesn't then you don't even call the backend but instead throw an PurchasedProductDetailsError.productNotInOrder. This approach in my opinion is better AND will be future resistant (for when we start supporting multiple products in orders).

Reply:
This also implies you have to delete the PurchaseProductDetailsQuery model because it has no use now.

Reply:
Added a TODO on the query.  This query is needed because ItineraryItem's currently don't contain a PurchasedProduct.  Once that's available, a PurchasedProduct can simply be passed in instead of a query as it would contain all the details necessary.